### DISCUSSION TITLE
How can I set http not https container registery?

### DISCUSSION THREAD

**Discussion by dameng324** [dotnet/sdk-container-builds#558] [Q&A] [answered]
My company has a local habor as internal container registery which use http only as only internal use.

but I set `<ConainterRegistery>` Property in my project and cannot push it. 
```
Encountered a SocketException with message "An existing connection was forcibly closed by the remote host.". Pausing before retry.
```

I guess the reason is default is https, can I set http protocol?

addition info:

I have already set docker `insecure-registries` config
```
  "insecure-registries": [
    "dockerhub.shengguanda.com"
  ]
```

If I use a Dockerfile and build with docker command, it will be fine.

**Comment by baronfel** [comment]
We don't support insecure registries yet, but adhering to docker configuration is probably a great way for us to go when we do support this. Is this something you'd be interested in helping contribute?

**Reply by dameng324** [reply]
I found the point here.

https://github.com/dotnet/sdk/blob/3a44947ed697a848c2ee667411e4b4127d3393fc/src/Containers/Microsoft.NET.Build.Containers/ContainerHelpers.cs#L150-L154

Check the local docker config, If  the registery is `localhost` or contained in `insecure-registeries` , will add `http` prefix, otherwise `https`.

I'm glad to help contribute this. but need some help, Is there any recommend api to get the local docker config?

**Reply by baronfel** [reply]
We don't yet have any code for reading the docker config file, because the only current use of that file is for authentication. Our use is in another library, https://github.com/mthalman/docker-creds-provider/blob/main/src%2FValleysoft.DockerCredsProvider%2FCredsProvider.cs, that hides this implementation detail. 

I think it would be ok to put an initial implementation into the SDK repo, then later we can decide if the Valleysoft library is the correct final destination for the code.

**Reply by dameng324** [reply]
It's really helpful. Thanks, I will try this.

**Reply by dameng324** [reply]
@baronfel  Hi, I use `DockerCli.GetDockerConfig()` to get the local docker config and create a PR to do this in this PR(https://github.com/dotnet/sdk/pull/39840).

**Comment by dameng324** [comment, marked as answer]
I create a nukpg for fix this problem. I can't push it to nuget.com because I dont want changed the package name. I just changed the version postfix.

If you run into this issue, you can download this package and put the nupkg file to your local nuget package source. 
Then change the Microsoft.NET.Build.Containers version to `8.0.203-sgd`
```xml
  <ItemGroup>
<!-- <PackageReference Include="Microsoft.NET.Build.Containers" Version="8.0.203" />-->
      <PackageReference Include="Microsoft.NET.Build.Containers" Version="8.0.203-sgd" />
  </ItemGroup>
```

It will work on 8.0.203 sdk version.

[Microsoft.NET.Build.Containers.8.0.203-sgd.nupkg.zip](https://github.com/dotnet/sdk-container-builds/files/14797951/Microsoft.NET.Build.Containers.8.0.203-sgd.nupkg.zip)
